src/components/WalletModal/index.tsx

```diff
@@ -150,6 +153,20 @@ export default function WalletModal({
     }
   }, [pendingConnector, walletView])
 
+  // When new wallet is successfully set by the user, trigger logging of Amplitude analytics event.
+  useEffect(() => {
```

Comment:
i don't think it's ideal to have state/and effect just for logging but i can't quite think of how to avoid it for now

Reply:
yeah same, spoke w vig at length about this, i think based off of how web3react state/redux is set up this is the best way we've got for now.